src/StudyViewer/StudyDetails.jsx

```diff
@@ -0,0 +1,238 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withRouter } from 'react-router-dom';
+import _ from 'lodash';
+import { Space, Typography, Descriptions, message, Divider, Alert, Modal, List } from 'antd';
+import Button from '@gen3/ui-component/dist/components/Button';
+import { FileOutlined, FilePdfOutlined, LinkOutlined } from '@ant-design/icons';
+import { capitalizeFirstLetter, humanFileSize } from '../utils';
+import { userHasMethodOnResource } from '../authMappingUtils';
+import { studyViewerConfig, useArboristUI, requestorPath, userapiPath } from '../localconf';
+import { fetchWithCreds } from '../actions';
+import './StudyViewer.css';
+
+const { Paragraph } = Typography;
+
+const getLabel = (label) => {
+  if (!studyViewerConfig.fieldMapping || studyViewerConfig.fieldMapping.length === 0) {
+    return capitalizeFirstLetter(label);
+  }
+  const fieldMappingEntry = studyViewerConfig.fieldMapping.find(i => i.field === label);
+  if (fieldMappingEntry) {
+    return fieldMappingEntry.name;
+  }
+  return capitalizeFirstLetter(label);
+};
+
+class StudyDetails extends React.Component {
+  state = { downloadModalVisible: false };
+
+  onRequestAccess = () => {
+    const body = {
+      username: this.props.user.username,
+      resource_path: '/programs/TODO', // FIXME: fill this
+      resource_id: this.props.data.rowAccessorValue,
+      resource_display_name: this.props.data.title,
+    };
+    fetchWithCreds({
+      path: `${requestorPath}request`,
+      method: 'POST',
+      body: JSON.stringify(body),
+    }).then(
+      ({ data, status }) => {
+        if (data && data.redirect_url) {
+          // if a redirect is configured, Requestor returns a redirect URL
+          window.open(data.redirect_url);
+        } else if (status !== 201) {
+          message
+            .error(`Something went wrong when talking to Requestor service, status ${status}`, 3);
+        } else {
+          message
+            .info('Request has sent successfully but Requestor has not been configured to redirect', 3);
+        }
```

Comment:
i don't think we need this, it's okay for other Commons to not configure a redirect

Reply:
yeah just trying to think in a more general way to address this

Reply:
```
if status == 201:
    if data.redirect_url:
        redirect user
else:
    message.error
```